packages/react-components/react-carousel/library/src/components/useEmblaCarousel.ts

```diff
@@ -142,22 +189,27 @@ export function useEmblaCarousel(
     }
   });
 
+  const handleIndexChange = React.useCallback(() => {
```

Comment:
```
type EmblaEventHandler = Parameters<EmblaCarouselType['on']>[1];

// ...
```

and then we can do:

```
  const handleIndexChange: EmblaEventHandler = useEventCallback((_, eventType) => {
    const newIndex = emblaApi.current?.selectedScrollSnap() ?? 0;
    const slides = emblaApi.current?.slideNodes();
    const actualIndex = emblaApi.current?.internalEngine().slideRegistry[newIndex][0] ?? 0;

    // We set the active or first index of group on-screen as the selected tabster index
    slides?.forEach((slide, slideIndex) => {
      setTabsterDefault(slide, slideIndex === actualIndex);
    });
    setActiveIndex(newIndex);

    if (eventType === 'autoplay:select') {
      onAutoplayIndexChange?.(null, { event: null, type: 'autoplay', index: newIndex });
    }
  });
```

🐱

Reply:
Sadly the eventType is generic so we only see a 'select' event on autoplay index change